src/main/scala/mesosphere/marathon/api/v2/json/Formats.scala

```diff
@@ -1058,11 +1058,14 @@ trait AppAndGroupFormats {
 
   implicit lazy val GroupFormat: Format[Group] = (
     (__ \ "id").format[PathId] ~
-    (__ \ "apps").formatNullable[Set[AppDefinition]].withDefault(Group.defaultApps) ~
+    (__ \ "apps").formatNullable[Iterable[AppDefinition]].withDefault(Iterable.empty) ~
     (__ \ "groups").lazyFormatNullable(implicitly[Format[Set[Group]]]).withDefault(Group.defaultGroups) ~
     (__ \ "dependencies").formatNullable[Set[PathId]].withDefault(Group.defaultDependencies) ~
     (__ \ "version").formatNullable[Timestamp].withDefault(Group.defaultVersion)
-  ) (Group(_, _, _, _, _), unlift(Group.unapply))
+  ) (
+      (id, apps, groups, dependencies, version) =>
+        Group(id, apps.map(app => app.id -> app).toMap, groups, dependencies, version),
```

Comment:
replace ".toMap" with "(collection.breakOut)" which should build the map for you (and it avoids an intermediate List of Pairs)